app/models/inventory-batch.js

```diff
@@ -16,7 +16,7 @@ export default AbstractModel.extend({
     },
     inventoryItemTypeAhead: {
       presence: {
-        if(object) {
+        unless(object) {
```

Comment:
@mkly my bad -- I misread the code.... I translated it as a regular javascript expression, eg if object is defined, return the value from object.hasInvoiceItems() to decide if validation fires.

You were correct in using unless and I think it is fine to use it in the project.